hackathon_site/event/serializers.py

```diff
@@ -87,6 +86,40 @@ def update(self, instance: Profile, validated_data):
         # ProfileSerializer update function and we want to keep the logic between these two update functions separate.
         return serializers.ModelSerializer.update(self, instance, validated_data)
 
+    def create(self, validated_data):
+        current_user = self.context["request"].user
+        if hasattr(current_user, "profile"):
+            raise serializers.ValidationError("User already has profile")
+
+        try:
+            review_status = Review.objects.get(application__user=current_user).status
+            if review_status != "Accepted":
```

Comment:
Do we also need to check the RSVP status?

Reply:
good catch, yes we do!